python/rascaline/utils.py

```diff
@@ -20,3 +22,225 @@ def _call_with_growing_buffer(callback, initial=1024):
+    serve to help users convert existing workflows.
+
+    Notes
+    -----
+    - This function does validate the values in the hyperparameter
+    dictionary, and it is up to the user to check that they pass
+    valid entries to `rascaline`.
+    - Not all of these parameters are supported in rascaline,
+    and some will raise warnings.
+
+    Parameters
+    ----------
+
+    mode: string in ["librascal", "dscribe"]
+          We anticipate future support for mode=="quip" as well
+
+    hyperparameters: dictionary of hyperparameter keys and values.
+        For mode = `librascal`, the anticipated values are:
+            - `coefficient_subselection`
+            - `compute_gradients`
+            - `covariant_lambda`
+            - `cutoff_function_parameters`
+            - `cutoff_function_type`
+            - `cutoff_smooth_width`
+            - `expansion_by_species_method`
+            - `gaussian_sigma_constant`
+            - `gaussian_sigma_type`
+            - `global_species`
+            - `interaction_cutoff`
+            - `inversion_symmetry`
+            - `max_angular`
+            - `max_radial`
+            - `normalize`
+            - `optimization_args`
+            - `optimization`
+            - `radial_basis`
+            - `soap_type`
+        For mode = `dscribe`, the anticipated values are:
+            - `average`
+            - `crossover`
+            - `dtype`
+            - `nmax`
+            - `lmax`
+            - `periodic`
+            - `rbf`
+            - `rcut`
+            - `sigma`
+            - `sparse`
+            - `species`
+
+    """
+    new_hypers = {}
+
+    if mode == "librascal":
+        anticipated_hypers = [
+            "coefficient_subselection",
+            "compute_gradients",
+            "covariant_lambda",
+            "cutoff_function_parameters",
+            "cutoff_function_type",
+            "cutoff_smooth_width",
+            "expansion_by_species_method",
+            "gaussian_sigma_constant",
+            "gaussian_sigma_type",
+            "global_species",
+            "interaction_cutoff",
+            "inversion_symmetry",
+            "max_angular",
+            "max_radial",
+            "normalize",
+            "optimization_args",
+            "optimization",
+            "radial_basis",
+            "soap_type",
+        ]
+
+        if any([key not in anticipated_hypers for key in hyperparameters]):
+            raise ValueError(
+                "I do not know what to do with the following hyperparameter entries:\n\t".format(
+                    "\n\t".join(
+                        [
+                            key
+                            for key in hyperparameters
+                            if key not in anticipated_hypers
+                        ]
+                    )
+                )
+            )
+
+        new_hypers["atomic_gaussian_width"] = hyperparameters.pop(
+            "gaussian_sigma_constant", None
```

Comment:
librascal has a lot of default values: `gaussian_sigma_constant` defaults to 0.3, `cutoff_smooth_width` defaults to 0.5, …

Should this code use the librascal defaults when translating to rascaline?